packages/openapi3/test/union-schema.test.ts

```diff
@@ -682,6 +681,82 @@ worksFor(["3.0.0"], ({ diagnoseOpenApiFor, oapiForModel, openApiFor }) => {
         message: "Cannot have a union containing only null types.",
       });
     });
+
```

Comment:
so i am still confused by openapi3 nullable and allOf relation, does it also mean we technically have the same issue for models(are scalar not the only problematic ones)?

Reply:
yeah I get the original issue, but this has opened a can of worms.

How in openapi3 those 2 different?

```yaml
schemas:
  str:
     type: string
  MyModel:
     type: object
    properties: {...}
//--- why do we allow b but not a
  Other:
     a:
        type: string
        allOf:
          - $ref: '#/components/schemas/str'
        nullable: true
     b:
        type: object
        allOf:
          - $ref: '#/components/schemas/MyModel'
        nullable: true
```

Reply:
Looks like we have to add a new schema for model with `nullable: true`, otherwise we have to use `{not: anyof: [type: xxx, ...]}` 😢

Reply:
I feel like we can focus on scalar and create another pr to solve model issues